File: guides/micronaut-scope-types/java/src/main/java/example/micronaut/singleton/Robot.java
//tag::pkg[]
package example.micronaut.singleton;
//end::pkg[]

//tag::clazz[]
import io.micronaut.core.annotation.NonNull;
import jakarta.inject.Singleton;

import java.util.UUID;

@Singleton // <1>
public class Robot {
    @NonNull
    private final String serialNumber;

    public Robot() {
        serialNumber = UUID.randomUUID().toString();
    }

    @NonNull
    public String getSerialNumber() {
        return serialNumber;
    }
}
//end::clazz[]

File: guides/micronaut-scope-types/kotlin/src/main/kotlin/example/micronaut/singleton/Robot.kt
package example.micronaut.singleton
/*
//tag::pkg[]
package example.micronaut.singleton
//end::pkg[]
*/

//tag::clazz[]
import jakarta.inject.Singleton
import java.util.UUID

@Singleton // <1>
class Robot {
    private val serialNumber: String = UUID.randomUUID().toString()

    fun getSerialNumber(): String {
        return serialNumber
    }
}
//end::clazz[]